app/src/main/java/com/infomaniak/drive/ui/addFiles/CreateDropBoxFolderFragment.kt

```diff
@@ -67,11 +72,20 @@ class CreateDropBoxFolderFragment : CreateFolderFragment() {
                 if (currentPermission == ONLY_ME) {
                     findNavController().popBackStack(R.id.newFolderFragment, true)
                 } else {
-                    safeNavigate(
-                        CreateDropBoxFolderFragmentDirections.actionCreateDropBoxFolderFragmentToFileShareDetailsFragment(
-                            fileId = file.id, ignoreCreateFolderStack = true
-                        )
-                    )
+                    lifecycleScope.launch(Dispatchers.IO) {
+
+                        FileController.addFileTo(newFolderViewModel.currentFolderId.value!!, file)
```

Comment:
It's better to have this in the `ViewModel`, it will allow you to avoid `Dispatchers` and `isResume`